### ISSUE TITLE
Implement new Concept Exercise: extension-methods

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/csharp#1445]
This issue describes how to implement the `extension-methods` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the Concept of Extension Methods in C#.

## Learning objectives

- Know what extension methods are.
- Know how to define extension methods on reference and value types..

## Out of scope

- Assemblies.

## Concepts

This Concepts Exercise's Concepts are:

- `extension-methods`: know what extension methods are; know how to define extension methods on reference and value types.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `classes`: know what instance methods are.
- `namespaces`: know what namespaces are; know how to import namespaces.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Extension methods][extension-methods]: how to define extension methods.
- [Namespaces][namespaces]: how to define an import namespaces.

### After

- [Extension methods][extension-methods]: how to define extension methods.
- [Namespaces][namespaces]: how to define an import namespaces.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/languages/csharp/reference/README.md#reference-docs
[reference-doc]: https://github.com/exercism/v3/blob/master/reference/concepts/extension-methods.md
[reference-example]: https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer
[exercise-example]: https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/numbers-floating-point
[design-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/design.md
[config.json-example]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/numbers/.meta/config.json
[concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[extension-methods]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-implement-and-call-a-custom-extension-method
[namespaces]: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/namespaces/

**Comment by yzAlvin** [comment]
Has there been work done on this issue yet? Would be happy to give it a go or work with you on this ryanrobidou :)

I was thinking of extending IEnumerable<int> ? People might be familiar with Sum, so maybe we could write an extension method Product, Mean, Median and Mode?

Would it be better to extend other types too? Maybe we could extend string with IsQuestion (just check that the string ends with a ?)

**Comment by Grenkin1988** [comment]
I guess this should be closed?

**Comment by ErikSchierboom** [comment]
Yep!
